examples/scripts/reward_trainer_summarize.py

```diff
@@ -13,155 +13,217 @@
+                    "attention_mask": feature["attention_mask_chosen"],
+                }
+            )
+            merged_features.append(
+                {
+                    "input_ids": feature["input_ids_rejected"],
+                    "attention_mask": feature["attention_mask_rejected"],
+                }
+            )
+        batch = self.tokenizer.pad(
+            merged_features,
+            padding=self.padding,
+            max_length=self.max_length,
+            pad_to_multiple_of=self.pad_to_multiple_of,
+            return_tensors=self.return_tensors,
+        )
+        batch = {
+            "input_ids": batch["input_ids"],
+            "attention_mask": batch["attention_mask"],
+            "return_loss": True,
+        }
+        return batch
+
 
-# Step 2: Load the dataset and pre-process it
-tokenizer = AutoTokenizer.from_pretrained(args.model_name)
-tokenizer.pad_token_id = tokenizer.eos_token_id
-model.config.pad_token_id = model.config.eos_token_id
-train_dataset = load_dataset(args.dataset_name, split="train")
-
-
-# Tokenize chosen/rejected pairs of inputs
-# Adapt this section to your needs for custom datasets
-def preprocess_function(examples):
-    new_examples = {
-        "input_ids_chosen": [],
-        "attention_mask_chosen": [],
-        "input_ids_rejected": [],
-        "attention_mask_rejected": [],
-    }
-    for prompt, chosen, rejected in zip(examples["prompt"], examples["chosen"], examples["rejected"]):
-        tokenized_prompt = tokenizer(prompt + "\n\n", max_length=512, truncation=True)
-        tokenized_chosen = tokenizer(chosen, max_length=48, truncation=True)
-        tokenized_rejected = tokenizer(rejected, max_length=48, truncation=True)
-        new_examples["input_ids_chosen"].append(tokenized_prompt["input_ids"] + tokenized_chosen["input_ids"])
-        new_examples["attention_mask_chosen"].append(tokenized_prompt["attention_mask"] + tokenized_chosen["attention_mask"])
-        new_examples["input_ids_rejected"].append(tokenized_prompt["input_ids"] + tokenized_rejected["input_ids"])
-        new_examples["attention_mask_rejected"].append(tokenized_prompt["attention_mask"] + tokenized_rejected["attention_mask"])
-
-    return new_examples
-
-
-# Preprocess the dataset and filter out examples that are longer than args.max_length
-train_dataset = train_dataset.map(
-    preprocess_function,
-    batched=True,
-    # num_proc=16,
-)
 
-if args.eval_split == "none":
-    eval_dataset = None
-else:
-    eval_dataset = load_dataset(args.dataset_name, split=args.eval_split)
+def create_and_prepare_model(args):
 
-    eval_dataset = eval_dataset.map(
-        preprocess_function,
-        batched=True,
-        num_proc=4,
+    torch_dtype = torch.float32
+
+    model = AutoModelForSequenceClassification.from_pretrained(
+        args.model_name,
+        num_labels=1,
+        torch_dtype=torch_dtype,
     )
 
+    model.config.torch_dtype = torch_dtype
+    model.config.use_cache = True
```

Comment:
I didn't use gradient checkpointing so I set this to true but I think it should be `not gradient_checkpointing` officially